src/gt4py/next/type_system/type_info.py

```diff
@@ -259,6 +259,26 @@ def is_type_or_tuple_of_type(type_: ts.TypeSpec, expected_type: type | tuple) ->
     return all(isinstance(t, expected_type) for t in primitive_constituents(type_))
 
 
+def is_tuple_of_type(type_: ts.TypeSpec, expected_type: type | tuple) -> TypeGuard[ts.TupleType]:
```

Comment:
More a note than an actual suggestion: This function feels to specialized for me, but I said the same about `is_type_or_tuple_of_type` when it was introduced.